packages/components/src/internal/components/entities/ParentEntityEditPanel.tsx

```diff
@@ -74,35 +74,39 @@ export class ParentEntityEditPanel extends Component<Props, State> {
     }
 
     init = async (): Promise<void> => {
-        const { parentDataType } = this.props;
-        const { typeListingSchemaQuery } = parentDataType;
-
-        try {
-            const optionsMap = await getEntityTypeOptions(parentDataType);
-
-            const parentTypeOptions = optionsMap.get(typeListingSchemaQuery.queryName);
-            const originalParents = getInitialParentChoices(parentTypeOptions, parentDataType, this.getChildModel());
-
-            this.setState({
-                currentParents: originalParents,
-                loading: false,
-                originalParents,
-                parentTypeOptions,
-            });
-        } catch (reason) {
-            this.setState({
-                error: getActionErrorMessage(
-                    'Unable to load ' + parentDataType.descriptionSingular + ' data.',
-                    parentDataType.descriptionPlural,
-                    true
-                ),
-            });
+        const { parentDataTypes, childData } = this.props;
+
+        let parentTypeOptions = List<IEntityTypeOption>();
+        let originalParents = List<EntityChoice>();
+
+        for (const parentDataType of parentDataTypes) {
+            const { typeListingSchemaQuery } = parentDataType;
+
+            try {
+                const options = await getEntityTypeOptions(parentDataType);
```

Comment:
Something to consider. Performing these requests using async/await within a for loop will result in serial requests. I would recommend using `Promise.all` instead and awaiting the mapped results. See https://stackoverflow.com/a/37576787.

Reply:
Nice, updated.